src/pages/Domains/index.tsx

```diff
@@ -68,7 +67,7 @@ function Domains({ address }: DomainsProps) {
   return (
     <Container>
       <DomainsHeader>Your Identities</DomainsHeader>
-      {!emptyDomainArray.length && !domainArray.length ? (
+      {!data || data?.length === 0 ? (
```

Comment:
This can be written with a single check:

`data?.length !== 0 ? 'Data mapping section' : 'Buy domain section'`.